website/static/js/fangorn.js

```diff
@@ -1974,31 +1997,32 @@ var FGToolbar = {
                 }, 'Cancel Pending Uploads')
             );
         }
-        //multiple selection icons
+        // multiple selection icons
+        // Special cased to not show 'delete multiple' for github or published dataverses
         if(items.length > 1 && ctrl.tb.multiselected()[0].data.provider !== 'github' && ctrl.tb.options.placement !== 'fileview' && !(ctrl.tb.multiselected()[0].data.provider === 'dataverse' && ctrl.tb.multiselected()[0].parent().data.version === 'latest-published') ) {
-            // Special cased to not show 'delete multiple' for github or published dataverses
-            var showDelete = false;
-            var each, i, len;
-            // Only show delete button if user has edit permissions on at least one selected file
-            for (i = 0, len = items.length; i < len; i++) {
-                each = items[i];
-                if (each.data.permissions.edit && !each.data.isAddonRoot && !each.data.nodeType) {
-                    showDelete = true;
-                    break;
+            if (showDeleteMultiple(items)) {
+                var preprintPath = getPreprintPath(window.contextVars.node.isPreprint, window.contextVars.node.preprintFileId);
+                if (preprintPath && multiselectContainsPreprint(items, preprintPath)) {
+                    generalButtons.push(
+                        m.component(FGButton, {
+                            icon: 'fa fa-trash',
+                            tooltip: 'One of these files is a Preprint. You cannot delete Preprints, but you can upload a new version.',                                        
```

Comment:
If this also covers folders, then it should have the vague folder method. Also if it has a checked out file, it should all be handled here, right? Try to make this super-generic for any reason why a folder/file would not be deletable.